docs/index.asciidoc

```diff
@@ -337,3 +337,26 @@ NOTE: Client identity is not typically validated using SSL because the receiving
 include::{include_path}/{type}.asciidoc[]
 
 :default_codec!:
+
+==== Elasticsearch output
+When this input is used along with an {logstash-ref}/plugins-outputs-elasticsearch.html#plugins-outputs-elasticsearch[Elasticsearch output pluginh], you have to set the `document_id` in the output configuration.
+
+[source]
+----
+output {
+  elasticsearch {
+      ...
+      document_id => "%{[@metadata][_id]}"
+      ...
+  }
+}
+----
+
+* For {esf-name} v1.9.0 and below you have to rename the field `_id` to `@metadata._id` with a filter:
+[source]
+----
+filter {
+    mutate { rename => { "_id" => "[@metadata][_id]" } }
+}
```

Comment:
While mutate's `rename` operation _effectively_ becomes a no-op for a source field that isn't present (such as an `_id` that is _not_ present on ESF 1.10), it may be useful to suggest a conditional wrapper that only runs the filter if `[_id]` is present and `[@metadata][_id]` is _not_ present to avoid ambiguity in cases where both are present.

The following should work with events sent by the old 1.9.x while being fully-forward-compatible with 1.10 where the rename isn't needed:

```suggestion
filter {
    # support ESF < 1.10
    if [_id] and ![@metadata][_id] {
      mutate { rename => { "_id" => "[@metadata][_id]" } }
    }
}
```